sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_credential.py

```diff
@@ -0,0 +1,37 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+import six
+
+
+class SharedKeyCredential(object):
+    """Credential type used for authenticating the client
+    with a subscription key.
+
+    :param str subscription_key: The subscription key to your Text Analytics
+        or Cognitive Services account.
+    """
+    def __init__(self, subscription_key):
+        if not isinstance(subscription_key, six.string_types):
+            raise TypeError("Please provide the subscription key as a string.")
+        self._subscription_key = subscription_key
+
+    @property
+    def subscription_key(self):
+        """Returns the current value of subscription key.
+        """
+        return self._subscription_key
+
+    def set_subscription_key(self, key):
```

Comment:
I like update_credential a little better here (explained why [here](https://github.com/Azure/azure-sdk-for-net/pull/9433#discussion_r368318018)), but we'll go with whatever the guidelines land on.